src/Internal/utils.php

```diff
@@ -8,10 +8,15 @@
 
 function safeGetByName(?array $collection): Closure
 {
-    return fn (string $name) => is_array($collection) && isset($collection[$name]) && $collection[$name];
+    return fn (string $name) => (is_array($collection) && isset($collection[$name])) ? $collection[$name] : null;
```

Comment:
I think no need to check is `$collection` an array, check condition `isset($collection[$name])` should be enough

Reply:
if $collection argument of type array is optionable `?array` then there is need to check if it's an array.

Reply:
I've checked that quickly online and that and indeed, looks like even if we use `isset` against null values it will not throw an error. I will test in application and if it behave the same i change it

Reply:
@paulfcdd  Added a change. RFR